process/src/main/java/io/smallrye/common/process/AbnormalExitException.java

```diff
@@ -0,0 +1,172 @@
+    /**
+     * Set to {@code true} if the hard timeout was known to have elapsed.
+     */
+    private boolean hardTimeoutElapsed;
+    /**
+     * The captured lines of error output.
+     */
+    private List<String> errorOutput = List.of();
+
+    /**
+     * Constructs a new {@code AbnormalExitException} instance. The message is left blank ({@code null}), and no
+     * cause is specified.
+     */
+    public AbnormalExitException() {
+    }
+
+    /**
+     * Constructs a new {@code AbnormalExitException} instance with an initial message. No
+     * cause is specified.
+     *
+     * @param msg the message
+     */
+    public AbnormalExitException(final String msg) {
+        super(msg);
+    }
+
+    /**
+     * Constructs a new {@code AbnormalExitException} instance with an initial cause. If
+     * a non-{@code null} cause is specified, its message is used to initialize the message of this
+     * {@code ProcessException}; otherwise the message is left blank ({@code null}).
+     *
+     * @param cause the cause
+     */
+    public AbnormalExitException(final Throwable cause) {
+        super(cause);
+    }
+
+    /**
+     * Constructs a new {@code AbnormalExitException} instance with an initial message and cause.
+     *
+     * @param msg the message
+     * @param cause the cause
+     */
+    public AbnormalExitException(final String msg, final Throwable cause) {
+        super(msg, cause);
+    }
+
+    /**
+     * {@return the exit code of the process}
+     */
+    public int exitCode() {
+        return exitCode;
+    }
+
+    /**
+     * Set the exit code of the process.
+     *
+     * @param exitCode the exit code
+     */
+    public void setExitCode(final int exitCode) {
+        this.exitCode = exitCode;
+    }
+
+    /**
+     * {@return {@code true} if the soft timeout elapsed before the process exited}
+     */
+    public boolean softTimeoutElapsed() {
+        return softTimeoutElapsed;
+    }
+
+    /**
+     * Set whether the soft timeout elapsed before the process exited.
+     *
+     * @param softTimeoutElapsed {@code true} if the soft timeout elapsed before the process exited
+     */
+    public void setSoftTimeoutElapsed(final boolean softTimeoutElapsed) {
+        this.softTimeoutElapsed = softTimeoutElapsed;
+    }
+
+    /**
+     * {@return {@code true} if the hard timeout elapsed before the process exited}
+     */
+    public boolean hardTimeoutElapsed() {
+        return hardTimeoutElapsed;
+    }
+
+    /**
+     * Set whether the hard timeout elapsed before the process exited.
+     *
+     * @param hardTimeoutElapsed {@code true} if the hard timeout elapsed before the process exited
+     */
+    public void setHardTimeoutElapsed(final boolean hardTimeoutElapsed) {
+        this.hardTimeoutElapsed = hardTimeoutElapsed;
+    }
+
+    /**
+     * {@return the captured error output of the process execution, if any}
+     */
+    public List<String> errorOutput() {
```

Comment:
is this stderr only ? what about stdout?

Reply:
We *could* capture stdout here, but doing so would mean adding "tee" support broadly. This can be done, but would require more effort. It's also possible that we could introduce this in the future.